packages/core/src/helpers/dataHelpers.ts

```diff
@@ -74,10 +71,25 @@ export function filterDataPerClauses(
     carry.push(sorter)
     return carry
   }, [] as ISortByObjectSorter<[string, Record<string, any>]>[])
-  const entriesOrdered = orderBy.length ? sort(entries).by(by) : entries
+  entries = orderBy.length ? sort(entries).by(by) : entries
+  // startAfter
+  if (startAfter && orderBy) {
+    const orderByKeys = orderBy.map(([path]) => path)
+    const startAfterValues = Array.isArray(startAfter)
+      ? startAfter
+      : orderByKeys.map((key) => startAfter[key])
+    if (startAfterValues.length > orderByKeys.length) {
+      throw new Error('startAfter must have the same or smaller number of values than orderBy')
+    }
+    for (const [index, key] of orderByKeys.entries()) {
+      const value = startAfterValues[index]
+      if (value == null) continue
+      const startIndex = entries.findIndex(([docId, docData]) => getProp(docData, key) === value)
+      entries = entries.slice(startIndex + 1)
+    }
+  }
```

Comment:
Reference: https://firebase.google.com/docs/firestore/query-data/query-cursors#paginate_a_query

`startAfter` must be used with `orderBy`